k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -84,8 +84,47 @@ pub struct GetUpgradeArgs {}
 impl GetUpgradeArgs {
     ///  Upgrade the resources.
     pub async fn get_upgrade(&self, namespace: &str) {
-        println!("{namespace}")
-        // To be implemented
+        // Create resources for getting upgrade status
+        GetUpgradeResource::create_get_upgrade_resource(namespace).await;
+    }
+}
+
+pub struct GetUpgradeResource {
+    pub(crate) event: Api<Event>,
```

Comment:
Consider making it private and adding a getter for this `pub(crate) fn event(&self) ...`